update!: Add AI chat recommendations

I-I-IT opened the pull request:
Need feedback on criteria, including reasonable rate limit. For the header, I will try to find how to upload the assets.

Feel free to add how to run AI models locally, with LM studio for example.

Changes proposed in this PR:

-

<!-- SCROLL TO BOTTOM TO AGREE!:
Please use a descriptive title for your PR, it will be included in our changelog!

If you are making changes that you have a conflict of interest with, please
disclose this as well (this does not disqualify your PR by any means):

Conflict of interest contributions involve contributing about yourself,
family, friends, clients, employers, or your financial and other relationships.
Any external relationship can trigger a conflict of interest.
-->

<!-- Place an x in the boxes below, like: [x] -->
- [X] I have disclosed any relevant conflicts of interest in my post.
- [X] I agree to grant Privacy Guides a perpetual, worldwide, non-exclusive, transferable, royalty-free, irrevocable license with the right to sublicense such rights through multiple tiers of sublicensees, to reproduce, modify, display, perform, relicense, and distribute my contribution as part of this project.
- [X] I am the sole author of this work. <!-- Do not check this box if you are not -->
- [X] I agree to the [Community Code of Conduct](https://www.privacyguides.org/coc).

<!-- What's this? When you submit a PR, you keep the Copyright for the work you
are contributing. We need you to agree to the above terms in order for us to
publish this contribution to our website. -->

TechFanTheo commented:
Relevant discussion : https://discuss.privacyguides.net/t/ai-tool/14347/16

TechFanTheo commented on docs/ai-chatbots.md:
```diff
@@ -0,0 +1,79 @@
+title : "AI Chatbots"
+icon: file/file
+description: Unlike OpenAI ChatGPT and it's successors, those AI Chatbots do not train their models using your conversations.
+cover:file/file
```
I just think we should change the name of this category **AI Chat** as it is the standard name used by every AI product. Also, I personally prefer this image as it is more clear that this is "talking robots "
https://unsplash.com/photos/a-white-robot-with-blue-eyes-and-a-laptop--0xMiYQmk8g

rollsicecream reviewed (changes requested):
I don't know why but the page is entirely broken for me.
![Capture d’écran du 2024-04-24 22-20-00](https://github.com/privacyguides/privacyguides.org/assets/153316540/fce6a9c0-27a0-4f28-9314-5046aeec144a)
![Capture d’écran du 2024-04-24 22-20-06](https://github.com/privacyguides/privacyguides.org/assets/153316540/a133fee3-7536-4e30-95df-c4a00a9556f4)
![Capture d’écran du 2024-04-24 22-20-11](https://github.com/privacyguides/privacyguides.org/assets/153316540/9a264e38-8b93-4564-8551-80ff47caea50)

TechFanTheo commented:
![Screenshot Recommended AI Chatbots Private ChatGPT Alternatives - Privacy Guides](https://github.com/privacyguides/privacyguides.org/assets/78900789/c3501d10-44ae-4f06-bd35-f10cb59b8af5)
I commited a patch, it seem to work, although we still have to add logo, add the Kobold icon to the PG assets repo. I also have some concerns, that I outline on the [forum post](https://discuss.privacyguides.net/t/ai-tool/14347/29)

navalspectator commented on docs/ai-chatbots.md:
```diff
@@ -0,0 +1,102 @@
+cover: file/file
+---
+Since the release of ChatGPT by OpenAI in December 2022, talking with Largue Language Models (LLMs) has become common. It can help us write better, understand unfamiliar subjects or answer a wide range of questions. However, the developpers of those models need data to improve their model. Unfortunately, this include your conversations with the AI.
+
```
We can divide this into two parts: Cloud-based models and local models.

navalspectator commented on docs/ai-chatbots.md:
```diff
@@ -0,0 +1,102 @@
+</details>
+</div>
+Brave Leo supports a range of models, including open-source models from Meta and Mixtral and closed-source models from Anthropic. Brave host the open models and do not retain chat or train on the data from those models. For Claude models, Brave proxies your chats through their servers,and the agreement between Brave and Anthropic states that chats are to be deleted after 30 days and that 'persona ...
+
```
GPT4ALL
Website: https://gpt4all.io/
GPT4All is an awesome free-to-use, locally running, privacy-aware chatbot developed by nomic.ai. After you install the app and download the models you want to use, it does not require access to the internet. Depending on your device RAM and storage, you can use a variety of open source models. It is available on Linux, Windows, and MacOS.

Telemetry, and the use of your chats for training purposes are opt-in and we do not suggest to activate those.

There is also LLamaChat, but apart from its UI, I dont see any reason to suggest it: https://www.llamachat.app/

freddy-m commented:
Closing this PR until the discussion is approved, cc @privacyguides/team

I-I-IT commented:
> Closing this PR until the discussion is approved, cc @privacyguides/team

??? There is a discussion, and we are drafting this. I don't see need to close this PR.  Other, like this one https://github.com/privacyguides/privacyguides.org/pull/2381, are also not approved yet.

dngray commented:
This was mistakenly closed.

I-I-IT commented:
Thanks

I-I-IT commented:
I have made a lot of modfications and I believe it is now a good quality for wide release.

xe-3 commented on docs/ai-chatbots.md:
```diff
@@ -0,0 +1,161 @@
+ It is available directly on [DuckDuckGo](https://duckduckgo.com), or [duck.ai](https://duck.ai). It is also available on DDG's onion site.
+<div class="admonition danger" markdown>
+<p class="admonition-title">Proprietary models providers retain your chats</p>
+Anthropic and OpenAI stores chats for up to 30 days. Do not use those models for anything you wouldn't want the World to know about.
```
Changes to the language to make this admonition more consistent with the Anthropic admonition in the Brave Leo section:

```suggestion
We recommend against using the proprietary models from Anthropic or OpenAI because those providers keep chat history for up to 30 days.
```

xe-3 commented on docs/ai-chatbots.md:
```diff
@@ -0,0 +1,161 @@
+</div>
+<div class="admonition danger" markdown>
+<p class="admonition-title">Claude chats are kept for 30 days </p>
+We recommend against using Anthropic's Claude models because the company keeps a chat history for 30 days.
```
Changes to make this admonition more consistent with the admonition in the Duckduckgo section:

```suggestion
We recommend against using Anthropic's proprietary Claude models because Anthropic keeps chat history for up to 30 days.
```

xe-3 commented on docs/ai-chatbots.md:
```diff
@@ -0,0 +1,161 @@
+We recommend manually switching off the "Use page context for response" toggle for pages with PII.
+</div>
+<div class="admonition danger" markdown>
+<p class="admonition-title">Claude chats are kept for 30 days </p>
```
```suggestion
<p class="admonition-title">Proprietary model providers retain your chats</p>
```

xe-3 reviewed (changes requested):
This review pertains only to the introduction and the 'hosted providers' section. I did not have time to review the local models section, but I plan to.

Most of the changes I've suggested are relate to consistency between sections, rewordings, or small typos and grammar corrections.

I-I-IT commented:
Thanks @xe-3 
I messed up with this [commit](https://github.com/privacyguides/privacyguides.org/pull/2525/commits/95db16724f4646046cb340aa84b1004a0779ef9d) cause I needed to edit your recommendation, not create a new individual commit.

I-I-IT commented:
@xe-3 Dix you look at the rest ?

xe-3 commented:
> @xe-3 Dix you look at the rest ?

Sorryl. I was holding off briefly because some developments on the local side have been happening in recent weeks, and then I just got distracted and dropped the ball. No promises, but I think I can look it over today.

xe-3 commented on docs/ai-chatbots.md:
```diff
@@ -0,0 +1,165 @@
+
+To run basic AI models you will fortunately not need a high-end computer. A computer with 8GB of RAM will be sufficient to run a 7B-8B parameters model. One issue with local models is that they have a slower inference speed than their cloud conterparts. You can remedy this by using your GPU.
+
+ But what language models can you run ? Meta and Mistral release their best model under an "open" license, while Google and Microsoft only release small, less capable models. The two leaders in the field OpenAI and Anthropic do not release their model publicly.But you can't run models in their original form, you will need to run [quantized models](https://huggingface.co/docs/optimum/en/concept_gu ...
```
Suggested (optional) rewrite:
```suggestion
There are many permissively licensed models available which can be run locally. [Hugging Face](https://huggingface.co/models?library=gguf) is a platform which lets you browse, research, and download models in common formats like GGUF. In addition to big names like Mistral, Meta, Microsoft, and Google, there are many community models and 'fine-tunes' to choose from. For consumer grade hardware, it is generally recommended to use [quantized models](https://huggingface.co/docs/optimum/en/concept_guides/quantization) for the best balance between model quality and performance.
```

xe-3 reviewed (changes requested):
Some changes for you to consider, plus one typo.

rollsicecream commented:
> @rollsicecream could you look at the current preview ? It is quite good now, compared to a few months ago.

Yeah, you're right on that! Though just 3 things to fix: 

- The DuckDuckGo AI PNG doesn't seem like to have the best image quality on the world lol
- Same thing for Ollama
- On the DuckDuckGo AI recommendation card, "search engine providers" should be rendered as a link with Markdown but doesn't seem to be the case here. Seems like a formatting problem.

I-I-IT commented:
Thank you for your answers. Let me know what are the future steps that I will need to take.

> * The DuckDuckGo AI PNG doesn't seem like to have the best image quality on the world lol
> * Same thing for Ollama

Both are fixed, I think it was my png to svg comversion that compressed them. I now just put them in png.
> * On the DuckDuckGo AI recommendation card, "search engine providers" should be rendered as a link with Markdown but doesn't seem to be the case here. Seems like a formatting problem.
Is fixed

kimg45 commented on docs/ai-chatbots.md:
```diff
@@ -0,0 +1,175 @@
+- [:material-account-cash: Surveillance Capitalism](basics/common-threats.md#surveillance-as-a-business-model){ .pg-brown }
+- [:material-close-outline: Censorship](basics/common-threats.md#avoiding-censorship){ .pg-blue-gray }
+
+Since the release of **ChatGPT** in 2022, talking with **Large Language Models** (*LLMs*) has become common. **LLMs can help us** write better, understand unfamiliar subjects or answer a wide range of questions. However, the developers of those models **need data to improve their models**. Unfortunately, this often **includes your conversations** with the AI. Those privates conversations must be  ...
```
```suggestion
Since the release of **ChatGPT** in 2022, talking with **Large Language Models** (*LLMs*) has become common. **LLMs can help us** write better, understand unfamiliar subjects or answer a wide range of questions. However, the developers of those models **need data to improve their models**. Unfortunately, this often **includes your conversations** with the AI. Those private conversations must be stored, which introduces a risk of data breaches. Furthermore, there is a real possibility the LLM will leak your private information in future conversations with other users.
```

kimg45 commented on docs/ai-chatbots.md:
```diff
@@ -0,0 +1,175 @@
+---
+meta_title: "Recommended AI Chatbots : Private ChatGPT Alternatives - Privacy Guides"
```
```suggestion
meta_title: "Recommended AI Chats : Private ChatGPT Alternatives - Privacy Guides"
```

kimg45 commented on docs/ai-chatbots.md:
```diff
@@ -0,0 +1,175 @@
+
+The recommendations for online providers here do not train their models using your chats and do not retain your chats for more than 14 days, based on each service's privacy policy. However, there is **no guarantee** that these privacy policies are honored.
+
+When using cloud-based AI chatbots, be mindful of the personal information you share. Even if a service doesn't store your conversations, there's still a risk of sensitive data being exposed or misused. **Do not share sensitive information** related to health, finance, or other highly personal matters to protect your privacy and security.
```
```suggestion
When using cloud-based AI chats, be mindful of the personal information you share. Even if a service doesn't store your conversations, there's still a risk of sensitive data being exposed or misused. **Do not share sensitive information** related to health, finance, or other highly personal matters to protect your privacy and security.
```

kimg45 commented on docs/tools.md:
```diff
@@ -479,6 +479,22 @@ For encrypting your operating system drive, we typically recommend using whichev
 </div>
 
 [Learn more :material-arrow-right-drop-circle:](frontends.md)
+### AI chatbots
```
```suggestion
### AI Chats
```

I-I-IT commented:
So I changed ai-chatbots/ai-chats to AI chat per Google Trends 
The files are now ai-chat, and the appeation is AI chat tools
![google-trends-ai](https://github.com/user-attachments/assets/ffd86988-c449-4131-8fe2-4b32af6b4ad5)

@rollsicecream I fixed the issues you told me. Is it good for you nows?

I-I-IT commented:
@ph00lt0 thanks for your suggestions. Please note that I have been working on a few changes to make the recommendations more digest and easy to understand for newcomers. It might be a couple of days before I push it, then I will review your changes.

I-I-IT commented:
@jonaharagon could you review this PR ? I have spent a lot of times working on this.

rollsicecream commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,175 @@
+
+- Should not retain your chats.
+- Should be accessible anonymously (Tor Browser Safest security level).
+- Providers should only have open-weight models that they self-host.
```
You can see the definition of that word by hovering over it with a computer mouse: "An open-weights model is an AI model that anyone can download and use, but for which the underlying training data and/or algorithms are proprietary."

rollsicecream commented:
> @rollsicecream I fixed the issues you told me. Is it good for you now?

That's pretty good :+1:

rollsicecream commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,175 @@
+- Should not retain your chats.
+- Should be accessible anonymously (Tor Browser Safest security level).
+- Providers should only have open-weight models that they self-host.
+- Should not be rate-limited.
```
Should be noted that Brave Leo, in its free plan, is rate-limited (5 msgs only I guess last time I checked)

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+[:octicons-home-16: Homepage](https://duck.ai){ .md-button .md-button--primary }
+[:simple-torbrowser:](https://duckduckgogg42xjoc72x3sjasowoarfbgcmvfimaftt6twagswzczad.onion/chat){ .card-link title="Onion Service" }
+[:octicons-eye-16:](https://duckduckgo.com/aichat/privacy-terms){ .card-link title="Privacy Policy" }
+[:octicons-info-16:](https://help.duckduckgo.com){ .card-link title=Documentation}
```
```suggestion
[:octicons-info-16:](https://help.duckduckgo.com){ .card-link title="Documentation" }
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+Duck has agreements with the third-party providers they use guaranteeing that they will not use your data for training their AI models. Proprietary model providers can keep a chat history for up to 30 days. For open-weights model, Duck uses the together.ai AI cloud platform, and has disabled history for those chats.
+
+It is available directly on [DuckDuckGo](https://duckduckgo.com), [duck.ai](https://duck.ai) or [DuckDuckGo onion site](https://duckduckgogg42xjoc72x3sjasowoarfbgcmvfimaftt6twagswzczad.onion/chat).
```
```suggestion
It is available directly on [DuckDuckGo](https://duckduckgo.com), [duck.ai](https://duck.ai), or [DuckDuckGo onion site](https://duckduckgogg42xjoc72x3sjasowoarfbgcmvfimaftt6twagswzczad.onion/chat).
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+<div class="admonition recommendation" markdown>
+
+![Brave Logo](assets/img/ai-chat/leo.svg){align=right}
```
```suggestion
![Brave Logo](assets/img/ai-chat/leo.svg){ align="right" }
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+[:octicons-home-16: Homepage](https://brave.com/leo){ .md-button .md-button--primary }
+[:octicons-eye-16:](https://brave.com/privacy/browser/#brave-leo){ .card-link title="Privacy Policy" }
+[:octicons-info-16:](https://github.com/brave/brave-browser/wiki/Brave-Leo){ .card-link title=Documentation}
```
```suggestion
[:octicons-info-16:](https://github.com/brave/brave-browser/wiki/Brave-Leo){ .card-link title="Documentation" }
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+## Local AI
+
+**Running AI models locally** offer a more private and secure alternative to cloud-based solutions, as **your data never leaves your device** and is therefore never shared with third-party providers. This provides peace of mind and **allows you to share sensitive information**.
```
```suggestion
**Running AI models locally** offers a more private and secure alternative to cloud-based solutions, as **your data never leaves your device** and is therefore never shared with third-party providers. This provides peace of mind and **allows you to share sensitive information**.
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+<div class="admonition recommendation" markdown>
+
+![Kobold.cpp Logo](assets/img/ai-chat/kobold.svg){align=right}
```
```suggestion
![Kobold.cpp Logo](assets/img/ai-chat/kobold.svg){ align="right" }
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+Kobold.cpp is an AI client that runs locally on your Windows, Mac, or Linux computer.
+
+In addition to supporting a large range of text models, Kobold.cpp also has a suports an image generators such as [Stable Diffusion](https://stability.ai/stable-image), and automatic speech recognition tools, such as [Whisper](https://github.com/ggerganov/whisper.cpp).
```
```suggestion
In addition to supporting a large range of text models, Kobold.cpp also supports image generators such as [Stable Diffusion](https://stability.ai/stable-image), and automatic speech recognition tools, such as [Whisper](https://github.com/ggerganov/whisper.cpp).
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+In addition to supporting a large range of text models, Kobold.cpp also has a suports an image generators such as [Stable Diffusion](https://stability.ai/stable-image), and automatic speech recognition tools, such as [Whisper](https://github.com/ggerganov/whisper.cpp).
+
+[:octicons-home-16: Homepage](https://github.com/LostRuins/koboldcpp){ .md-button .md-button--primary }
+[:octicons-info-16:](https://github.com/LostRuins/koboldcpp/wiki){ .card-link title=Documentation}
```
```suggestion
[:octicons-info-16:](https://github.com/LostRuins/koboldcpp/wiki){ .card-link title="Documentation" }
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+</div>
+
+Kobold shines best when you are looking for heavy customisation and tweaking, such as for roleplaying purposed. It allows you to modify the model temperature,the context window. It also supports creating a network tunnel to access AI models from other devices, such as your phone.
```
```suggestion
Kobold shines best when you are looking for heavy customization and tweaking, such as for roleplaying purposes. It allows you to modify the model temperature from the context window. It also supports creating a network tunnel to access AI models from other devices, such as your phone.
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+</div>
+
+Kobold shines best when you are looking for heavy customisation and tweaking, such as for roleplaying purposed. It allows you to modify the model temperature,the context window. It also supports creating a network tunnel to access AI models from other devices, such as your phone.
+
```
```suggestion
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+Ollama is an easy-to-use command-line AI assistant that is available on macOS, Linux, and Windows (preview).
+
+In addition to supporting a wide range of text models, Ollama also supports [LLaVA](https://github.com/haotian-liu/LLaVA) models and has *experimental* support for Meta's[Llama vision capabilities](https://huggingface.co/blog/llama32#what-is-llama-32-vision).
+
```
```suggestion
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+<div class="admonition recommendation" markdown>
+
+![Ollama Logo](assets/img/ai-chat/ollama.png){align=right}
```
```suggestion
![Ollama Logo](assets/img/ai-chat/ollama.png){ align="right" }
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+Ollama is an easy-to-use command-line AI assistant that is available on macOS, Linux, and Windows (preview).
+
+In addition to supporting a wide range of text models, Ollama also supports [LLaVA](https://github.com/haotian-liu/LLaVA) models and has *experimental* support for Meta's[Llama vision capabilities](https://huggingface.co/blog/llama32#what-is-llama-32-vision).
```
```suggestion
In addition to supporting a wide range of text models, Ollama also supports [LLaVA](https://github.com/haotian-liu/LLaVA) models and has *experimental* support for Meta's [Llama vision capabilities](https://huggingface.co/blog/llama32#what-is-llama-32-vision).
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+
+[:octicons-home-16: Homepage](https://github.com/ollama/ollama){ .md-button .md-button--primary }
+[:octicons-info-16:](https://github.com/ollama/ollama){ .card-link title=Documentation}
```
```suggestion
[:octicons-info-16:](https://github.com/ollama/ollama){ .card-link title="Documentation" }
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+</div>
+
+Ollama shines best when you are looking for an AI client that has great compatibility and ease of use. It runs on all desktop platforms and doesn't involve any manual setup, while still using inference and other techniques to make outputs faster.
```
```suggestion
Ollama shines best when you're looking for an AI client that has great compatibility and ease of use. It runs on all desktop platforms and doesn't involve any manual setup, while still using inference and other techniques to make outputs faster.
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+Ollama shines best when you are looking for an AI client that has great compatibility and ease of use. It runs on all desktop platforms and doesn't involve any manual setup, while still using inference and other techniques to make outputs faster.
+
+It also simplifies the process of setting up a Local AI chat setup, as it downloads the AI model you want to use automatically. For example "ollama run llama3.2" will automatically download and run the Llama 3.2 model.
```
```suggestion
It also simplifies the process of setting up a local AI chat, as it downloads the AI model you want to use automatically. For example "ollama run llama3.2" will automatically download and run the Llama 3.2 model.
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+<div class="admonition recommendation" markdown>
+
+![Llamafile Logo](assets/img/ai-chat/llamafile.svg){align=right}
```
```suggestion
![Llamafile Logo](assets/img/ai-chat/llamafile.svg){ align="right" }
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+![Llamafile Logo](assets/img/ai-chat/llamafile.svg){align=right}
+
+Llamafile is a single-file executable that allows users to run large language models locally on their own computers, without any setup involved. It is available on Linux, macOS and Windows.
+
+The Mozilla-run project also supports LLaVA. However, it does not support speech recognition or image generation. Finally, it is very lightweight and compatible.
```
```suggestion
Llamafile is a single-file executable that allows users to run large language models locally on their own computers, without any setup involved. It is available on Linux, macOS, and Windows.

The Mozilla-run project also supports LLaVA. However, it does not support speech recognition or image generation. It's also very lightweight and compatible.
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+The Mozilla-run project also supports LLaVA. However, it does not support speech recognition or image generation. Finally, it is very lightweight and compatible.
+
+[:octicons-home-16: Homepage](https://github.com/Mozilla-Ocho/llamafile/){ .md-button .md-button--primary }
+[:octicons-info-16:](https://github.com/Mozilla-Ocho/llamafile/?tab=readme-ov-file#llamafile){ .card-link title=Documentation}
```
```suggestion
[:octicons-info-16:](https://github.com/Mozilla-Ocho/llamafile/?tab=readme-ov-file#llamafile){ .card-link title="Documentation" }
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+[:octicons-home-16: Homepage](https://github.com/Mozilla-Ocho/llamafile/){ .md-button .md-button--primary }
+[:octicons-info-16:](https://github.com/Mozilla-Ocho/llamafile/?tab=readme-ov-file#llamafile){ .card-link title=Documentation}
+[:octicons-code-16:](https://github.com/ollama/ollama){ .card-link title="Source Code"}
```
```suggestion
[:octicons-code-16:](https://github.com/ollama/ollama){ .card-link title="Source Code" }
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+[:octicons-home-16: Homepage](https://github.com/Mozilla-Ocho/llamafile/){ .md-button .md-button--primary }
+[:octicons-info-16:](https://github.com/Mozilla-Ocho/llamafile/?tab=readme-ov-file#llamafile){ .card-link title=Documentation}
+[:octicons-code-16:](https://github.com/ollama/ollama){ .card-link title="Source Code"}
+[:octicons-lock-16:](https://github.com/Mozilla-Ocho/llamafile?tab=readme-ov-file#security){ .card-link title="Security Policy"}
```
```suggestion
[:octicons-lock-16:](https://github.com/Mozilla-Ocho/llamafile?tab=readme-ov-file#security){ .card-link title="Security Policy" }
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+<div class="admonition note" markdown>
+<p class="admonition-title">Few Models Available</p>
+Mozilla has only made available llamafiles for some Llama and Mistral models, while there are few third-party llamafiles available. Another issue is that Windows limits .exe files to 4GB size, and most models are larger than that. To fix both of those issues, you can load external weights, as detailed [here](https://github.com/Mozilla-Ocho/llamafile?tab=readme-ov-file#using-llamafile-with-externa ...
```
```suggestion
Mozilla has only made llamafiles available for some Llama and Mistral models, while there are few third-party llamafiles available. Another issue is that Windows limits .exe files to 4GB, and most models are larger than that. To fix both of those issues, you can load external weights, as detailed [here](https://github.com/Mozilla-Ocho/llamafile?tab=readme-ov-file#using-llamafile-with-external-weights).
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+#### Cloud Providers
+
+- The provider or third-parties they use must not use your chats for training
```
```suggestion
- The provider or third-parties they use must not use your chats for training.
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+- Must provide at least one model with high rate limits, to allow an user to use it for medium to heavy workloads.
+
+#### Local AI clients
+- Must be open-source
```
```suggestion
- Must be open-source.
```

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,233 @@
+
+#### Local AI clients
+
+- Should be easy to download and setup, such as having a one-click install process.
```
```suggestion
- Should be easy to download and set up, such as having a one-click install process.
```

friadev commented on docs/tools.md:
```diff
@@ -469,6 +469,26 @@ For encrypting your operating system drive, we typically recommend using whichev
+
+<div class="grid cards" markdown>
+
+- ![Duck AI logo](assets/img/ai-chat/duckai.png){ .twemoji loading=lazy }[Duck AI](ai-chat.md#duckduckgo-ai-chat)
```
```suggestion
- ![Duck AI logo](assets/img/ai-chat/duckai.png){ .twemoji loading="lazy" }[Duck AI](ai-chat.md#duckduckgo-ai-chat)
```

friadev commented on docs/tools.md:
```diff
@@ -469,6 +469,26 @@ For encrypting your operating system drive, we typically recommend using whichev
+<div class="grid cards" markdown>
+
+- ![Duck AI logo](assets/img/ai-chat/duckai.png){ .twemoji loading=lazy }[Duck AI](ai-chat.md#duckduckgo-ai-chat)
+- ![Leo AI logo](assets/img/ai-chat/leo.svg){ .twemoji loading=lazy }[Brave Leo](ai-chat.md#brave-leo)
```
```suggestion
- ![Leo AI logo](assets/img/ai-chat/leo.svg){ .twemoji loading="lazy" }[Brave Leo](ai-chat.md#brave-leo)
```

friadev commented on docs/tools.md:
```diff
@@ -469,6 +469,26 @@ For encrypting your operating system drive, we typically recommend using whichev
+
+<div class="grid cards" markdown>
+
+- ![Kobold logo](assets/img/ai-chat/kobold.svg){ .twemoji loading=lazy }[Kobold.cpp](ai-chat.md#koboldcpp)
```
```suggestion
- ![Kobold logo](assets/img/ai-chat/kobold.svg){ .twemoji loading="lazy" }[Kobold.cpp](ai-chat.md#koboldcpp)
```

friadev commented on docs/tools.md:
```diff
@@ -469,6 +469,26 @@ For encrypting your operating system drive, we typically recommend using whichev
+<div class="grid cards" markdown>
+
+- ![Kobold logo](assets/img/ai-chat/kobold.svg){ .twemoji loading=lazy }[Kobold.cpp](ai-chat.md#koboldcpp)
+- ![Llamafile logo](assets/img/ai-chat/llamafile.svg){ .twemoji loading=lazy }[Llamafile](ai-chat.md#llamafile)
```
```suggestion
- ![Llamafile logo](assets/img/ai-chat/llamafile.svg){ .twemoji loading="lazy" }[Llamafile](ai-chat.md#llamafile)
```

friadev commented on docs/tools.md:
```diff
@@ -469,6 +469,26 @@ For encrypting your operating system drive, we typically recommend using whichev
+
+- ![Kobold logo](assets/img/ai-chat/kobold.svg){ .twemoji loading=lazy }[Kobold.cpp](ai-chat.md#koboldcpp)
+- ![Llamafile logo](assets/img/ai-chat/llamafile.svg){ .twemoji loading=lazy }[Llamafile](ai-chat.md#llamafile)
+- ![Ollama logo](assets/img/ai-chat/ollama.png){ .twemoji loading=lazy }[Ollama](ai-chat.md#ollama)
```
```suggestion
- ![Ollama logo](assets/img/ai-chat/ollama.png){ .twemoji loading="lazy" }[Ollama](ai-chat.md#ollama)
```

friadev commented on includes/abbreviations.en.txt:
```diff
@@ -1,4 +1,5 @@
 *[2FA]: 2-Factor Authentication
+*[AI]: Artificial Intelligence
 *[ADB]: Android Debug Bridge
 *[AOSP]: Android Open Source Project
 *[ATA]: Advanced Technology Attachment
```
```suggestion
*[ADB]: Android Debug Bridge
*[AI]: Artificial Intelligence
*[AOSP]: Android Open Source Project
*[ATA]: Advanced Technology Attachment
```

friadev commented on includes/abbreviations.en.txt:
```diff
@@ -71,6 +75,7 @@
 *[PGP]: Pretty Good Privacy (see OpenPGP)
 *[PII]: Personally Identifiable Information
 *[QNAME]: Qualified Name
+*[rate limits]:Rate limits are restrictions that a service imposes on the number of times a user can access their services within a specified period of time.
```
```suggestion
*[rate limits]: Rate limits are restrictions that a service imposes on the number of times a user can access their services within a specified period of time.
```

I-I-IT commented:
Thanks @friadev for suggestions. I implemented most of them, and commented for those I don't think are needed. BTW, I also added a table to quickly glance at the different AI chats.

friadev commented on docs/ai-chat.md:
```diff
@@ -0,0 +1,273 @@
+
+</div>
+
+Kobold shines best when you are looking for heavy customisation and tweaking, such as for roleplaying purposed. It allows you to modify parameters such as the AI model temperature and the AI chat's system prompt. It also supports creating a network tunnel to access AI models from other devices, such as your phone.
```
```suggestion
Kobold shines best when you are looking for heavy customization and tweaking, such as for roleplaying purposes. It allows you to modify parameters such as the AI model temperature and the AI chat's system prompt. It also supports creating a network tunnel to access AI models from other devices, such as your phone.
```

friadev commented on docs/tools.md:
```diff
@@ -468,6 +468,26 @@ For encrypting your operating system drive, we typically recommend using whichev
+- ![Ollama logo](assets/img/ai-chat/ollama.png){ .twemoji loading=lazy }[Ollama](ai-chat.md#ollama)
+</div>
+
+
```
```suggestion
```

friadev commented on includes/abbreviations.en.txt:
```diff
@@ -62,6 +64,7 @@
 *[OCSP]: Online Certificate Status Protocol
 *[OEM]: Original Equipment Manufacturer
 *[OEMs]: Original Equipment Manufacturers
+*[open-weights]: An open weights model is an AI model that anyone can download and use, but for which the underlying training data and/or algorithms are proprietary.
```
```suggestion
*[open-weights]: An open weights-model is an AI model that anyone can download and use, but for which the underlying training data and/or algorithms are proprietary.
```

jonaharagon commented on theme/assets/img/cover/ai-chatbots.webp:
Where is this image from? We need a source link and licensing information to publish it.

friadev commented:
It's from unsplash: https://unsplash.com/photos/a-white-robot-with-blue-eyes-and-a-laptop--0xMiYQmk8g

jonaharagon commented on theme/assets/img/ai-chat/ollama.svg:
This isn't an actual SVG image, it appears to be a PNG embedded in an SVG file. This is not going to be well compatible with browsers (and is also just bad), so we need to replace it with the official https://ollama.com/public/ollama.png, or find a real vector file, or make our own (which I think @dngray has done before and knows how to do?)

jonaharagon commented:
Thanks for the PR @I-I-IT! These changes are now going to be tracked separately here:

- #2810 
- #2811
